internal/testutil/tc/advanced_cluster.go

```diff
@@ -195,3 +197,124 @@ func checkShardedOldSchemaMultiCloud(name string, numShards int, analyticsSize s
+		CheckDestroy:             acc.CheckDestroyCluster,
+		PreCheck:                 acc.PreCheckBasicSleep(t, nil, projectID, clusterName),
+		Steps: []resource.TestStep{
+			{
+				Config: configTenant(projectID, clusterName),
+				Check:  checkTenant(projectID, clusterName),
+			},
+			{
+				Config: configTenant(projectID, clusterNameUpdated),
+				Check:  checkTenant(projectID, clusterNameUpdated),
+			},
+			acc.TestStepImportCluster(resourceName),
+		},
+	}
+}
+
+func configTenant(projectID, name string) string {
+	return fmt.Sprintf(`
+		resource "mongodbatlas_advanced_cluster" "test" {
+			project_id   = %[1]q
+			name         = %[2]q
+			cluster_type = "REPLICASET"
+
+			replication_specs = [{
+				region_configs = [{
+					electable_specs = {
+						instance_size = "M5"
+					}
+					provider_name         = "TENANT"
+					backing_provider_name = "AWS"
+					region_name           = "US_EAST_1"
+					priority              = 7
+				}]
+			}]
+		}
+	`, projectID, name)
+}
+
+func checkTenant(projectID, name string) resource.TestCheckFunc {
+	attrsSet := []string{"replication_specs.#", "replication_specs.0.id", "replication_specs.0.region_configs.#"}
+	attrsMap := map[string]string{
+		"project_id":                           projectID,
+		"name":                                 name,
+		"termination_protection_enabled":       "false",
+		"global_cluster_self_managed_sharding": "false",
+		"labels.#":                             "0",
+	}
+	checks := acc.AddAttrSetChecks(resourceName, nil, attrsSet...)
+	checks = acc.AddAttrChecks(resourceName, checks, attrsMap)
+	return resource.ComposeAggregateTestCheckFunc(checks...)
+}
+
+func TenantUpgrade(t *testing.T, projectID, clusterName string) *resource.TestCase {
+	t.Helper()
+	return &resource.TestCase{
+		PreCheck:                 acc.PreCheckBasicSleep(t, nil, projectID, clusterName),
+		ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
+		CheckDestroy:             acc.CheckDestroyCluster,
+		Steps: []resource.TestStep{
+			{
+				Config: acc.ConvertAdvancedClusterToTPF(t, configTenant(projectID, clusterName)),
+				Check:  checkTenant(projectID, clusterName),
+			},
+			{
+				Config: acc.ConvertAdvancedClusterToTPF(t, configTenantUpgraded(projectID, clusterName)),
+				Check:  checksTenantUpgraded(projectID, clusterName),
+			},
+		},
+	}
+}
+
+func configTenantUpgraded(projectID, name string) string {
+	return fmt.Sprintf(`
+	resource "mongodbatlas_advanced_cluster" "test" {
+		project_id   = %[1]q
+		name         = %[2]q
+		cluster_type = "REPLICASET"
+		
+		replication_specs {
+			region_configs {
+				priority        = 7
+				provider_name = "AWS"
+				region_name     = "US_EAST_1"
+				electable_specs {
+					node_count = 3
+					instance_size = "M10"
+				}
+			}
+		}
+	}
+	`, projectID, name)
+}
+
+func enableChecksLatestTpf(checkMap map[string]string) map[string]string {
+	newMap := map[string]string{}
+	for k, v := range checkMap {
```

Comment:
@lantoli prototype of what we were discussing about checks

Reply:
@EspenAlbert that's great, thanks.  ping me when this is merged so i'll use the idea in adv_cluster tests as well